core/src/main/java/edu/softserve/zoo/exceptions/ApplicationException.java

```diff
@@ -0,0 +1,143 @@
+package edu.softserve.zoo.exceptions;
+
+/**
+ * This is a common superclass of all exceptions thrown by application.
+ * This exception and its subclasses are unchecked exceptions.
+ * {@code ApplicationException} class includes builder for exception instances.
+ *
+ * @author Julia Siroshtan
+ */
+public abstract class ApplicationException extends RuntimeException {
+
+    private static final long serialVersionUID = -5774091581747092157L;
```

Comment:
I've decided to add proper serialization support because this class is implicitly serializable (its parent Throwable implements Serializable interface). I'll remove it if we don't need it :)